core/src/data/dataSource.h

```diff
@@ -18,6 +18,68 @@ class TileManager;
 struct RawCache;
 class Texture;
 
+struct RawDataSource {
+    virtual bool loadTileData(std::shared_ptr<TileTask> _task, TileTaskCb _cb) = 0;
+
+    /* Stops any running I/O tasks pertaining to @_tile */
+    virtual void cancelLoadingTile(const TileID& _tile) {
+        if (next) { next->cancelLoadingTile(_tile); }
+    }
+
+    virtual void clear() { if (next) next->clear(); }
+
+
+    std::unique_ptr<RawDataSource> next;
+};
+
+class NetworkDataSource : public RawDataSource {
```

Comment:
Downloading bundles should be handled separately in my opinion, similar to what we do in ClientGeoJson source or become part of scene loading.